### ISSUE TITLE
PDF417 reader early returns with multiple codes on a page

### ISSUE THREAD

**Issue by ezekiel-buck** [zxing/zxing#1764]
I have a test case that is drawing barcodes on a page (below) and confirming the reader can read the 2 codes in various conditions 

```
        String barcodeText = "Barcode One";
	String barcodeText2 = "Barcode Two";
	BufferedImage image = new BufferedImage(1000, 1500, BufferedImage.TYPE_BYTE_BINARY);
	BarCode barCode = BarCodeFactory.getBarCode(barcodeFormat, barcodeText, 400, 50);
	BarCode barCode2 = BarCodeFactory.getBarCode(barcodeFormat, barcodeText2, 400, 50);

	java.awt.Graphics2D g = image.createGraphics();
	g.setColor(Color.WHITE);
	g.fillRect(0, 0, image.getWidth(), image.getHeight());
	g.setColor(Color.BLACK);
	// print barcodes with 50px room from top and bottom
	barCode.draw(g, 100, 100);
	barCode2.draw(g, 100, 1400);
```

The code to read the barcodes:

```
        Hashtable<DecodeHintType, Object> hints = new Hashtable<>();
	hints.put(DecodeHintType.TRY_HARDER, Boolean.TRUE);
	hints.put(DecodeHintType.POSSIBLE_FORMATS, possibleFormats);
	hints.put(DecodeHintType.ALSO_INVERTED, Boolean.TRUE);
	BinaryBitmap binaryBitmap = new BinaryBitmap(new HybridBinarizer(luminanceSource));

	try {
	        MultiFormatReader reader = new MultiFormatReader();
	        MultipleBarcodeReader multipleBarcodeReader = new GenericMultipleBarcodeReader(reader);
		barCodes = multipleBarcodeReader.decodeMultiple(binaryBitmap, hints);
	} catch (NotFoundException ex) {
		 // Ignore, we will handle 0 results later
	}
```

This code detects both barcodes when the 'barcodeFormat' is Code39 or Code128. However, PDF417 codes are not being read once the first one is found.

**Comment by srowen** [comment]
The "this barcode isn't detected" isn't an actionable issue

**Comment by ezekiel-buck** [comment]
It IS detecting the first 417 code. However, it is early returning and not even attempting to read for the second. I have confirmed this behavior by moving the "second" barcode closer to the center so that it is read first (without moving the first barcode that I confirmed was being read) and the reader doesn't read the "first" code that WAS previously being read.

This was to prove to myself that I hadn't drawn the codes too close to the edge and got a corrupted code.
